### PR TITLE
WIP for testing: SAML Config Refactor

### LINKED ISSUE #2950: SimpleSAML PHP Fatal Error: Cannot redeclare ah_db_current_host
My system information:
* BLT version: 9.1

Issue:
Running SimpleSAML PHP and attempting to login trips:

php-error web-2585 [13-Jul-2018 17:00:58 UTC] PHP Fatal error: Cannot redeclare ah_db_current_host() (previously declared in /vendor/simplesamlphp/simplesamlphp/config/acquia_config.php:205) in /vendor/simplesamlphp/simplesamlphp/config/acquia_config.php on line 205 

Likely because we define this in our acquia config AND it's in the simplesamlphp module: https://github.com/acquia/blt/blob/9.x/scripts/simplesamlphp/acquia_config.php#L192

Proposed resolution is to remove our copy of the ah_db_current_host in the acquia config file in BLT.

### PR DESCRIPTION
**Pull request by lcatlett**
Fixes #2950 #2949 

Changes proposed:
- WIP for testing: Refactors acquia config to remove problematic functions in favor dynmically generating base config

### CODE REVIEW

**Comment by ba66e77** [comment]
Is this going to subsume any of the other SAML PR's?

**Comment by mikemadison13** [comment]
@ba66e77  i think if this one goes in we will no longer need:
https://github.com/acquia/blt/pull/2949
https://github.com/acquia/blt/pull/2948

**Comment by mikemadison13** [comment]
https://github.com/acquia/blt/pull/2924 is still needed i believe

File: scripts/simplesamlphp/acquia_config.php [hunk 1]
```diff
@@ -54,156 +52,52 @@
...
+  $config['store.type'] = 'sql';
+  $config['store.sql.dsn'] = sprintf('mysql:host=%s;port=%s;dbname=%s', '127.0.0.1', '', 'drupal');
+  $config['store.sql.username'] = 'drupal';
+  $config['store.sql.password'] = 'drupal';
+  $config['store.sql.prefix'] = 'simplesaml';
+  $config['certdir'] = "/var/www/simplesamlphp/cert/";
+  $config['metadatadir'] = "/var/www/simplesamlphp/metadata";
+  $config['baseurlpath'] = 'simplesaml/';
+  $config['loggingdir'] = '/var/www/simplesamlphp/log/';
 }
```
Comment (mikemadison13) [inline scripts/simplesamlphp/acquia_config.php]: there are 2 } here at lines 68/69 and only 1 is needed (fatal)

File: scripts/simplesamlphp/acquia_config.php [hunk 2]
```diff
@@ -54,156 +52,52 @@
...
-    ]);
-    $response = $resolver->query("cluster-{$db_cluster_id}.mysql", 'CNAME');
-    $cached_id = $response->answer[0]->cname;
-  }
-  catch (\Net_DNS2_Exception $e) {
-    $cached_id = '';
-  }
-
-  return $cached_id;
 }
```
Comment (mikemadison13) [inline scripts/simplesamlphp/acquia_config.php]: there are 3 } here and only 1 is needed (fatal)

**Comment by lcatlett** [comment]
closing in favor of #2953

**Comment by mikemadison13** [comment]
@lcatlett looking again, it's not i don't think. looks like you covered the multi-site stuff in this. sorry!

**Comment by lcatlett** [comment]
@ba66e77 This PR should subsume any other SAML config PRs.

File: scripts/simplesamlphp/acquia_config.php [hunk 3]
```diff
@@ -38,172 +28,91 @@
...
- *   The name of a MySQL database.
- *
- * @return array
- *   Database information.
- */
-function db_info($db_name) {
-  $creds_json = file_get_contents('/var/www/site-php/' . getenv('AH_SITE_NAME') . '/creds.json');
+elseif (getenv('AH_SITE_ENVIRONMENT')) {
+  // Set  ACE ad ACSF sites based on hosting database and site name.
+  $config['certdir'] = "/mnt/www/html/{$_ENV['AH_SITE_GROUP']}.{$_ENV['AH_SITE_ENVIRONMENT']}/simplesamlphp/cert/";
```
Comment (rjgwiz) [inline scripts/simplesamlphp/acquia_config.php]: `/mnt/www/html/{$_ENV['AH_SITE_GROUP']}.{$_ENV['AH_SITE_ENVIRONMENT']}/simplesamlphp/` is pointing to the `simplesamlphp` symlink in the docroot from `symlinkDocrootToLibDir()` right? Using a relative path works and would make non-environment specific, i.e. `$config['certdir'] = 'cert/';`. Same for `$config['metadatadir']` on the next line

Comment (lcatlett) [reply scripts/simplesamlphp/acquia_config.php]: @rjgwiz I believe the simplesamlphp_auth module has a note that it requires absolute paths for those files - have you found that not to be the case?

Comment (rjgwiz) [reply scripts/simplesamlphp/acquia_config.php]: @lcatlett yea - in our project we have been using relative path for over a year. Currently on [simplesamlphp](https://github.com/simplesamlphp/simplesamlphp) `v1.15.4`

**Comment by ba66e77** [comment]
@mikemadison13, you had requested changes on this PR. Have they all been addressed at this point?

**Comment by mikemadison13** [comment]
@ba66e77 yes i think we are good now!

**Pull request merged**